ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -1084,7 +1095,8 @@ static bool InitializeEmptyPlaceholder(const char* fullPath, TPlaceholder* data,
 static bool IsVirtualizationRoot(const char* fullPath)
 {
     PrjFSVirtualizationRootXAttrData data = {};
-    if (TryGetXAttr(fullPath, PrjFSVirtualizationRootXAttrName, sizeof(PrjFSVirtualizationRootXAttrData), &data))
+    if (TryGetXAttr(fullPath, PrjFSVirtualizationRootXAttrName, sizeof(PrjFSVirtualizationRootXAttrData), &data)
+        || TryGetXAttr(fullPath, PrjFSVirtualizationRootXAttrNameOld, sizeof(PrjFSVirtualizationRootXAttrData), &data))
```

Comment:
@wilbaker what was the final decision on if we wanted backwards compat?

Reply:
Since we don't have to support backwards compat right now, let's not take on maintenance costs of adding it.

What we could do, however, is bump the on-disk version such that on Windows there's a no-op upgrade and on Mac there is a message that a breaking change occurred and users must reclone.

(That could be its own PR as long as we don't ship before completing that work)

cc: @jrbriggs @nickgra

Reply:
We don't even need to bump the on-disk version for this, given that the xattr name is different. As soon as we detect the old xattr name, we can output that error.

Reply:
> We don't even need to bump the on-disk version for this, given that the xattr name is different. As soon as we detect the old xattr name, we can output that error.

Unfortunately once the repo is mounted VFS4G has no way to output errors to the user.

When we've had to make breaking changes like this in the past we've adjusted `RepoMetadata.DiskLayoutVersion.MinimumSupportedMajorVersion` so that when mounting a user will see:

>https://github.com/Microsoft/VFSForGit/blob/cf69c01dace22ddeef8e7bbc3a9322a1e73abfac/GVFS/GVFS.Common/DiskLayoutUpgrade.cs#L79) 

> As soon as we detect the old xattr name, we can output that error.


I don't think we should add the overhead of checking for the old xattr name.  If we make `MinimumSupportedMajorVersion` platform specific (and set it to a new value on Mac) we can prevent the old repos from mounting in the first place which eliminates the need to check for the old xattrs.

Reply:
> I don't think we should add the overhead of checking for the old xattr name. If we make `MinimumSupportedMajorVersion` platform specific (and set it to a new value on Mac) we can prevent the old repos from mounting in the first place which eliminates the need to check for the old xattrs.

I should have included that I think that change and this change should get merged into master at the same time, @jrbriggs, is there someone on our side with the bandwidth to help make this change in the VFS4G product code (unless you'd like to make the change @pmj)?

Reply:
@wilbaker, I think I can help with that if you are looking volunteers.  

Were you thinking just move MinimumSupportedMajorVersion to GVFSPlatform, and up to 8 for Mac?

Reply:
> @wilbaker, I think I can help with that if you are looking volunteers.

Ok great!

> Were you thinking just move MinimumSupportedMajorVersion to GVFSPlatform, and up to 8 for Mac?

I was thinking that at a minimum we'll want to move `MinimumSupportedMajorVersion` into  `GVFSPlatform`, but I hadn't dug into what other changes we'd need to make.

Currently the on-disk version for both platforms is 17 (the original idea was to keep them in sync if we could), but perhaps we should allow them to diverge, and have separate Mac and Windows platform on-disk version numbers.  In our current situation, for example, we need to be able to increase both the `MinimumSupportedMajorVersion` and the current major version on Mac without impacting either of those version numbers on Windows.

I think it's worth spending a bit of time to dig into the options here so that we can avoid painting ourselves into a corner.  Feel free to catch me offline if you want some additional details on the current state of the on-disk version code.

Reply:
> I should have included that I think that change and this change should get merged into master at the same time, @jrbriggs, is there someone on our side with the bandwidth to help make this change in the VFS4G product code (unless you'd like to make the change @pmj)?

I think @jeschu1 is no doubt better prepared for this task than I am; happy to help out wherever necessary/appropriate though!

@jeschu1 feel free to push to this PR branch so we can get both changes merged at the same time - I think that should work; if not, let me know.

Reply:
@pmj, I added the commit to separate versioning on Mac/Windows.

Reply:
Thanks @jeschu1 - they look sane to me but I'm not really the right person to be signing them off so I'll leave that to those more experienced with the VFS4G userland code.